raiden/smart_contracts/NettingChannelLibrary.sol

```diff
@@ -339,7 +329,7 @@ library NettingChannelLibrary {
         uint8 index = 1 - index_or_throw(self, caller_address);
 
         Participant storage counterparty = self.participants[index];
-        if (counterparty.nonce == 0) {
```

Comment:
This change is not mentioned in the PR's description.

I do recall that we had discussed the nonce check made little sense here but can you add a small explanation why you are changing it to `locksroot`?

Especially since 6 lines below we are again [checking](https://github.com/raiden-network/raiden/pull/527/files#diff-09cebda618db5fb2268a55de8de94279L348) the same locksroot against the root we calculate from the lock and the merkle proof.

Is it to save some gas by failing earlier and not going into `computeMerkleRoot()`? If so it's not needed since `throw` would eat up all provided gas anyway.